model/Storage.inc.php

```diff
@@ -277,10 +284,25 @@ public static function getUploadQueueItemsAndInfo($hash) {
 				'item'=> $item
 			];
 		}
-
 		return $results;
 	}
 
+	public static function succeedUploadsByHash($hash) {
```

Comment:
If we add a `processed` column, this can be `markUploadsAsProcessed` (and just a single SQL command). Deletion will be done by event scheduler.